src/cli/commands/lint.ts

```diff
@@ -124,8 +126,11 @@ async function lint(name: string, flags: any, command: Lint, rules?: RuleCollect
   if (flags.verbose) {
     command.log(`Linting ${name}`);
   }
-  const spec: IParserResult = await readParsable(name, flags.encoding);
-  const spectral = new Spectral();
+
+  const targetUri = resolve(name);
+
+  const spec: IParserResult = await readParsable(targetUri, flags.encoding);
+  const spectral = new Spectral({ resolver: httpAndFileResolver });
```

Comment:
One more thought.
I'd say the above might be quite a significant change. It may slow down CI since Spectral will start making HTTP requests. We should provide a way to opt-out - a CLI flag here for resolver would be more than welcome.
It's a different story, though, so it can stay as is for now.

Reply:
It will only make http requests if there are http $refs in the spec, and if there are http $refs in the spec, we can't really validate the spec without resolving them.

Reply:
I'm aware, but IMHO we should provide a way to opt-out.

Reply:
What would that actually mean though, would it just fail for folks using the CLI if they have HTTP refs? I think then opt out is removing the HTTP refs from your spec right?

Reply:
> What would that actually mean though, would it just fail for folks using the CLI if they have HTTP refs? I think then opt out is removing the HTTP refs from your spec right?

Opting-out would mean turning off http/https $ref resolution.
Resolving external resources may significantly slow down CI. Not a big deal though, we can leave it as is.

Reply:
Yeah opting out would mean it would fail for folks if they use http $refs. I can't see a use case for this - how about we re-consider if somebody brings it up later?

Reply:
Yeah we've had a lot of complaints that it cant resolve and I cant imagine it going the other way. I agree we should reconsider if somebody asks for a way to turn it off. Maybe for security reasons or something else.